lib/postgres_to_redshift.rb

```diff
@@ -128,14 +128,36 @@ def import_table(table)
     puts "Importing #{table.target_table_name}"
     target_connection.exec("DROP TABLE IF EXISTS public.#{table.target_table_name}_updating")
 
-    target_connection.exec("BEGIN;")
+    begin
+      target_connection.exec('BEGIN;')
+
+      target_connection.exec("ALTER TABLE public.#{target_connection.quote_ident(table.target_table_name)} RENAME TO #{table.target_table_name}_updating")
+
+      target_connection.exec("CREATE TABLE public.#{target_connection.quote_ident(table.target_table_name)} (#{table.columns_for_create})")
+
+      target_connection.exec("COPY public.#{target_connection.quote_ident(table.target_table_name)} FROM 's3://#{ENV['S3_DATABASE_EXPORT_BUCKET']}/export/#{table.target_table_name}.psv.gz' CREDENTIALS 'aws_access_key_id=#{ENV['S3_DATABASE_EXPORT_ID']};aws_secret_access_key=#{ENV['S3_DATABASE_EXPORT_KEY']}' GZIP TRUNCATECOLUMNS ESCAPE DELIMITER as '|';")
+
+      target_connection.exec('COMMIT;')
 
-    target_connection.exec("ALTER TABLE public.#{target_connection.quote_ident(table.target_table_name)} RENAME TO #{table.target_table_name}_updating")
+    rescue PG::InternalError => e
+      target_connection.exec('ROLLBACK;')
 
-    target_connection.exec("CREATE TABLE public.#{target_connection.quote_ident(table.target_table_name)} (#{table.columns_for_create})")
+      print_last_redshift_loading_error if e.message.include?('stl_load_errors')
```

Comment:
It seems like this will not print any errors that do not match. Will other kinds of errors fail silently?